swap/cheque.go

```diff
@@ -94,7 +97,7 @@ func (cheque *Cheque) Equal(other *Cheque) bool {
 		return false
 	}
 
-	if cheque.CumulativePayout != other.CumulativePayout {
+	if !cheque.CumulativePayout.Equals(other.CumulativePayout) {
```

Comment:
Maybe better `if ...Equals` return `true`?

Reply:
i don't think we can claim that 2 cheques are equal only because their cumulative payouts are equal, no?

Reply:
Of course you are right, I just looked at the truncated diff and didn't see that there are other checks afterwards